src/plugins/data/common/search/utils.ts

```diff
@@ -11,17 +11,36 @@ import { AggTypesDependencies } from '..';
 import type { IKibanaSearchResponse } from './types';
 
 /**
+ * From https://github.com/elastic/elasticsearch/issues/55572: "When is_running is false, the query has stopped, which
+ * may happen due to ... the search failed, in which case is_partial is set to true to indicate that any results that
+ * may be included in the search response come only from a subset of the shards that the query should have hit."
  * @returns true if response had an error while executing in ES
  */
 export const isErrorResponse = (response?: IKibanaSearchResponse) => {
-  return !response || !response.rawResponse || (!response.isRunning && !!response.isPartial);
+  return (
+    !response ||
+    !response.rawResponse ||
+    (!response.isRunning &&
+      !!response.isPartial &&
+      // See https://github.com/elastic/elasticsearch/pull/97731. For CCS with ccs_minimize_roundtrips=true, isPartial
+      // is true if the search is complete but there are shard failures. In that case, the _clusters.details section
+      // will have information about those failures. This will also likely be the behavior of CCS with
+      // ccs_minimize_roundtrips=false and non-CCS after https://github.com/elastic/elasticsearch/issues/98913 is
+      // resolved.
+      !response.rawResponse?._clusters?.details)
+  );
 };
 
 /**
  * @returns true if response is completed successfully
  */
 export const isCompleteResponse = (response?: IKibanaSearchResponse) => {
-  return Boolean(response && !response.isRunning && !response.isPartial);
+  // Some custom search strategies do not indicate whether they are still running. In this case, assume it is complete.
+  if (response && !response.hasOwnProperty('isRunning')) {
+    return true;
+  }
+
+  return !isErrorResponse(response) && Boolean(response && !response.isRunning);
```

Comment:
Thx for fixing it so quickly 👍 ! QQ: from consumer view like here:

https://github.com/elastic/kibana/blob/c97d4960bf995671c7053333236ae7ff47fcc7a4/src/plugins/discover/public/application/main/utils/fetch_documents.ts#L55-L70

We can be sure, that now with this change , no partial results are declared as complete accidentally, because we can rely on `isRunning`, right?

Reply:
Yes, that's correct.

Reply:
thx!